### PR TITLE
feat(web-renderer): add EnvironmentMonitor and ServerState

### PR DESCRIPTION
**Pull request by AngeloFilaseta**
This PR contains the EnvironmentMonitor implementation. The monitor maps the Environment to its surrogate class and stores it in the Server state, which is also added in this PR.

### CODE REVIEW

**Comment by DanySK** [comment]
How does `ServerState` work?

File: alchemist-web-renderer/src/jvmMain/kotlin/it/unibo/alchemist/server/state/ServerState.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import it.unibo.alchemist.common.model.surrogate.EnvironmentSurrogate
+import it.unibo.alchemist.common.model.surrogate.PositionSurrogate
+import it.unibo.alchemist.common.state.CommonState
+import it.unibo.alchemist.core.interfaces.Simulation
+import it.unibo.alchemist.server.state.reducers.environmentSurrogateReducer
+import it.unibo.alchemist.server.state.reducers.simulationReducer
+
+/**
+ * The state of the server.
```
Comment (DanySK) [inline alchemist-web-renderer/src/jvmMain/kotlin/it/unibo/alchemist/server/state/ServerState.kt:20]: @AngeloFilaseta can you document this class better, linking to the pattern and explaining the rationale behind the singleton state, like you did in the discussion comment?

**Review by DanySK — CHANGES_REQUESTED** [review]

File: alchemist-web-renderer/src/jvmMain/kotlin/it/unibo/alchemist/server/state/ServerState.kt (hunk 1 above)
Comment (AngeloFilaseta) [reply alchemist-web-renderer/src/jvmMain/kotlin/it/unibo/alchemist/server/state/ServerState.kt:20]: Perfect, I'll do the same thing for the ClientState.

Comment (DanySK) [reply alchemist-web-renderer/src/jvmMain/kotlin/it/unibo/alchemist/server/state/ServerState.kt:20]: OK. Please resolve this conversation once done, and the PR will automerge.

**Pull request merged**